boot/bootutil/src/loader.c

```diff
@@ -956,21 +961,34 @@ boot_validated_swap_type(struct boot_loader_state *state,
         if(reset_addr < PM_CPUNET_B0N_ADDRESS)
 #endif
         {
+            const struct flash_area *nsib_fa;
             const struct flash_area *primary_fa;
             rc = flash_area_open(flash_area_id_from_multi_image_slot(
-                        BOOT_CURR_IMG(state),
-                        BOOT_PRIMARY_SLOT),
-                    &primary_fa);
-
+                                 BOOT_CURR_IMG(state), BOOT_PRIMARY_SLOT),
+                                 &primary_fa);
             if (rc != 0) {
                 return BOOT_SWAP_TYPE_FAIL;
             }
-            /* Get start and end of primary slot for current image */
-            if (reset_addr < primary_fa->fa_off ||
-                    reset_addr > (primary_fa->fa_off + primary_fa->fa_size)) {
-                /* The image in the secondary slot is not intended for this image
-                */
-                return BOOT_SWAP_TYPE_NONE;
+
+            if (reset_addr < primary_fa->fa_off) {
+                    /* NSIB upgrade slot */
+                    rc = flash_area_open((uint32_t)_image_1_primary_slot_id,
+                                    &nsib_fa);
+
+                    if (rc != 0) {
+                            return BOOT_SWAP_TYPE_FAIL;
+                    }
+
+                    /* Image is placed before Primary and within the NSIB slot */
+                    if (reset_addr > nsib_fa->fa_off
+                                    && reset_addr < (nsib_fa->fa_off + nsib_fa->fa_size)) {
+                            /* Set primary to be NSIB upgrade slot */
+                            BOOT_IMG_AREA(state, 0) = nsib_fa;
+                    /* Check start and end of primary slot for current image */
+                    }
+	    }    else if (reset_addr > (primary_fa->fa_off + primary_fa->fa_size)) {
+                        /* The image in the secondary slot is not intended for any */
+                            return BOOT_SWAP_TYPE_NONE;
```

Comment:
alignment here is completely off

Reply:
Fixed seems that a tabexpand went wrong sorry.